src/System.Private.ServiceModel/tools/SelfHostedWcfService/Program.cs

```diff
@@ -0,0 +1,282 @@
+            WebSocketHttpDuplexBinaryBufferedTestServiceHost webSocketHttpDuplexBinaryBufferedTestServiceHostServiceHost = new WebSocketHttpDuplexBinaryBufferedTestServiceHost(typeof(WcfService.WSDuplexService), webSocketHttpDuplexBinaryBufferedTestServiceHostbaseAddress);
+            webSocketHttpDuplexBinaryBufferedTestServiceHostServiceHost.Open();
+
+            Uri[] webSocketHttpsRequestReplyBinaryBufferedTestServiceHostbaseAddress = new Uri[]  { new Uri(string.Format("{0}/WebSocketHttpsRequestReplyBinaryBuffered.svc", websocketsBaseAddress))};
+            WebSocketHttpsRequestReplyBinaryBufferedTestServiceHost webSocketHttpsRequestReplyBinaryBufferedTestServiceHostServiceHost = new WebSocketHttpsRequestReplyBinaryBufferedTestServiceHost(typeof(WcfService.WSRequestReplyService), webSocketHttpsRequestReplyBinaryBufferedTestServiceHostbaseAddress);
+            webSocketHttpsRequestReplyBinaryBufferedTestServiceHostServiceHost.Open();
+
+            Uri[] webSocketHttpsRequestReplyTextBufferedTestServiceHostbaseAddress = new Uri[]  { new Uri(string.Format("{0}/WebSocketHttpsRequestReplyTextBuffered.svc", websocketsBaseAddress))};
+            WebSocketHttpsRequestReplyTextBufferedTestServiceHost webSocketHttpsRequestReplyTextBufferedTestServiceHostServiceHost = new WebSocketHttpsRequestReplyTextBufferedTestServiceHost(typeof(WcfService.WSRequestReplyService), webSocketHttpsRequestReplyTextBufferedTestServiceHostbaseAddress);
+            webSocketHttpsRequestReplyTextBufferedTestServiceHostServiceHost.Open();
+
+            Uri[] webSocketHttpsDuplexBinaryBufferedTestServiceHostbaseAddress = new Uri[]  { new Uri(string.Format("{0}/WebSocketHttpsDuplexBinaryBuffered.svc", websocketsBaseAddress))};
+            WebSocketHttpsDuplexBinaryBufferedTestServiceHost webSocketHttpsDuplexBinaryBufferedTestServiceHostServiceHost = new WebSocketHttpsDuplexBinaryBufferedTestServiceHost(typeof(WcfService.WSDuplexService), webSocketHttpsDuplexBinaryBufferedTestServiceHostbaseAddress);
+            webSocketHttpsDuplexBinaryBufferedTestServiceHostServiceHost.Open();
+
+            Uri[] webSocketHttpsDuplexTextBufferedTestServiceHostbaseAddress = new Uri[]  { new Uri(string.Format("{0}/WebSocketHttpsDuplexTextBuffered.svc", websocketsBaseAddress))};
+            WebSocketHttpsDuplexTextBufferedTestServiceHost webSocketHttpsDuplexTextBufferedTestServiceHostServiceHost = new WebSocketHttpsDuplexTextBufferedTestServiceHost(typeof(WcfService.WSDuplexService), webSocketHttpsDuplexTextBufferedTestServiceHostbaseAddress);
+            webSocketHttpsDuplexTextBufferedTestServiceHostServiceHost.Open();
+
+            Console.WriteLine("All service hosts have started.");
+            Console.WriteLine("Press <ENTER> to terminate the self service Host.");
+            Console.ReadLine();
```

Comment:
You will get an unhandled exception with call stacks, thus it's obvious what's going wrong.